v3/src/components/graph/utilities/pixi-transition.ts

```diff
@@ -10,109 +10,73 @@ const defaultInterpolation = smoother
 // const pow2out: Interpolation = (x: number) => Math.pow(x - 1, 2) * (-1) + 1
 
-export type SupportedPropKey = "position" | "scale"
-export type SupportedPropValue = { x: number, y: number }
+export type TransitionProp = "position" | "scale"
+
+export type TransitionTarget = { x: number, y: number, transition: PixiTransition }
+
+export type TransitionPropMap = Partial<Record<TransitionProp, Map<PIXI.Sprite, TransitionTarget>>>
 
 export class PixiTransition {
+  startTime = performance.now()
   duration = 0
-  time = 0
-  isFinished = false
-  frameId: number | undefined
-  points: PIXI.Sprite[] = []
   onEndCallback?: () => void
 
-  targetProp: Partial<Record<SupportedPropKey, Map<PIXI.Sprite, SupportedPropValue>>> = {}
-  startProp: Partial<Record<SupportedPropKey, Map<PIXI.Sprite, SupportedPropValue>>> = {}
-
-  constructor(duration: number, points: PIXI.Sprite[]) {
-    this.duration = duration
-    this.points = points
-  }
-
-  setTargetPosition(point: PIXI.Sprite, x: number, y: number) {
-    this.setTargetXyProp("position", point, x, y)
-  }
-
-  setTargetScale(point: PIXI.Sprite, scale: number) {
-    this.setTargetXyProp("scale", point, scale, scale)
-  }
-
-  play() {
-    const transitionProps = Object.keys(this.targetProp) as SupportedPropKey[]
-    if (transitionProps.length === 0) {
-      this.handleOnEnd()
-      return
+  static anyTransitionActive(targetProps: TransitionPropMap) {
+    const transitionProps = Object.keys(targetProps) as TransitionProp[]
+    for (const propName of transitionProps) {
+      if (targetProps[propName]?.size) {
+        return true
+      }
     }
+    return false
+  }
 
-    let time = 0
-    const duration = this.duration
-    const startTime = performance.now()
-
-    const transitionFrame = () => {
-      const timeRatio = Math.min(1, time / duration)
-      const factor = defaultInterpolation(timeRatio)
-
-      transitionProps.forEach(propKey => {
-        this.xyTransition(propKey, factor)
-      })
+  static transitionStep(targetProps: TransitionPropMap, startProps: TransitionPropMap) {
+    const now = performance.now()
+    const finishedTransitions = new Set<PixiTransition>()
+    const transitionProps = Object.keys(targetProps) as TransitionProp[]
 
-      if (time < duration) {
-        this.frameId = requestAnimationFrame(transitionFrame)
-        time = performance.now() - startTime
-      } else {
-        this.frameId = undefined
-        this.handleOnEnd()
+    transitionProps.forEach(propKey => {
+      const targetProp = targetProps[propKey]
+      const startProp = startProps[propKey]
+      if (!targetProp || !startProp) {
+        return
       }
-    }
 
-    transitionFrame()
-  }
-
-  handleOnEnd() {
-    this.onEndCallback?.()
-    this.isFinished = true
-  }
+      for (const [point, target] of targetProp.entries()) {
+        const transition = target.transition
+        const factor = transition.getFactor(now)
```

Comment:
As suggested elsewhere, perhaps `progress` would be more intuitive than `factor`.

Reply:
Thanks, I changed it to progress in https://github.com/concord-consortium/codap/pull/1060/commits/225f14e744826088a1a4009ea5f6ddc9d134a54c.